File: backend/app/schemas/analytics.py
"""Pydantic schemas for analytics and dashboard API responses."""

from pydantic import BaseModel


class DashboardStats(BaseModel):
    """Top-level dashboard statistics."""

    total_jobs_found: int = 0
    total_applications: int = 0
    applications_pending: int = 0
    applications_applied: int = 0
    applications_interview: int = 0
    applications_rejected: int = 0
    applications_offer: int = 0
    avg_ats_score: float = 0.0
    total_llm_cost_usd: float = 0.0


class ApplicationFunnelData(BaseModel):
    """Application funnel stage counts."""

    stage: str
    count: int


class ATSScoreDistribution(BaseModel):
    """ATS score histogram bucket."""

    range_label: str
    count: int


class LLMUsageStats(BaseModel):
    """LLM usage aggregation."""

    provider: str
    model: str
    total_requests: int = 0
    total_tokens: int = 0
    total_cost_usd: float = 0.0
    avg_latency_ms: float = 0.0


class TimelineEntry(BaseModel):
    """Daily activity timeline entry."""

    date: str
    applications_created: int = 0
    applications_applied: int = 0
    jobs_found: int = 0

File: frontend/src/types/analytics.ts
/** Top-level dashboard statistics. */
export interface DashboardStats {
  total_jobs_found: number;
  total_applications: number;
  applications_pending: number;
  applications_applied: number;
  applications_interview: number;
  applications_rejected: number;
  applications_offer: number;
  avg_ats_score: number;
  total_llm_cost_usd: number;
}

/** A single stage in the application funnel. */
export interface ApplicationFunnelData {
  stage: string;
  count: number;
}

/** ATS score histogram bucket. */
export interface ATSScoreDistribution {
  range_label: string;
  count: number;
}

/** LLM provider usage aggregation. */
export interface LLMUsageStats {
  provider: string;
  model: string;
  total_requests: number;
  total_tokens: number;
  total_cost_usd: number;
  avg_latency_ms: number;
}

/** Daily activity timeline entry. */
export interface TimelineEntry {
  date: string;
  applications_created: number;
  applications_applied: number;
  jobs_found: number;
}
